pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -93,31 +118,40 @@ type ParamType string
 const (
 	ParamTypeString ParamType = "string"
 	ParamTypeArray  ParamType = "array"
+	ParamTypeObject ParamType = "object"
 )
 
 // AllParamTypes can be used for ParamType validation.
-var AllParamTypes = []ParamType{ParamTypeString, ParamTypeArray}
+var AllParamTypes = []ParamType{ParamTypeString, ParamTypeArray, ParamTypeObject}
 
 // ArrayOrString is modeled after IntOrString in kubernetes/apimachinery:
 
 // ArrayOrString is a type that can hold a single string or string array.
 // Used in JSON unmarshalling so that a single JSON field can accept
 // either an individual string or an array of strings.
+// TODO: This struct will be renamed or be embedded in a new struct to take into
```

Comment:
> Looks like we are going to rename the struct based on the next resolved comment - any reason to not do it in this PR itself?

I think @ywluogg has sent a headsup email to the community asking for inputs. So we think it might be better to do this a bit later to make sure this rename (or creating new struct) doesn't break others who are using the old struct name?

Reply:
Ok that's fine

Reply:
> I think @ywluogg has sent a headsup email to the community asking for inputs. So we think it might be better to do this a bit later to make sure this rename (or creating new struct) doesn't break others who are using the old struct name.

I created this [issue](https://github.com/tektoncd/pipeline/issues/4858) for following up the issue